lib/plaid/user.rb

```diff
@@ -332,11 +334,17 @@ def delete
     # Returns another User record with the same access token, but tied to the
     # new product.
     def upgrade(product)
+      User.upgrade product, access_token, client: client
+    end
+
+    # Internal: Upgrade a User instance.
+    def self.upgrade(product, access_token, client: nil)
```

Comment:
I don't like this method. It's sole purpose is to workaround the `private_class_method :new` which you introduce in the same commit.

I think checking the product in the constructor would be enough. And `private_class_method :new` isn't needed at all.